libsplinter/src/biome/notifications/database/models.rs

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 Cargill Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------------
+ */
+
+use std::time::SystemTime;
+
+use super::schema::{notifications, user_notifications};
+
+#[derive(Insertable, Queryable)]
+#[table_name = "notifications"]
+pub struct Notification {
+    pub id: i64,
+    pub payload_title: String,
+    pub payload_body: String,
+    pub created: SystemTime,
+    pub origin: String,
+    pub link: String,
+    pub metadata: serde_json::Value,
```

Comment:
I'm concerned about this field being JSON. I think it should probably just be key/value data (string/string?).

Reply:
The more I think about this, the more I think this should be called 'properties' and put in a notification_properties table with notification_id,key,value.